bundles/org.openhab.voice.openaitts/src/main/java/org/openhab/voice/openaitts/internal/OpenAITTSService.java

```diff
@@ -0,0 +1,149 @@
+import org.osgi.service.component.annotations.Modified;
+import org.osgi.service.component.annotations.Reference;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonObject;
+
+/**
+ * @author Artur Fedjukevits - Initial contribution
+ *         API documentation: https://platform.openai.com/docs/guides/text-to-speech
+ */
+@Component(configurationPid = TTS_SERVICE_PID, property = Constants.SERVICE_PID + "="
+        + TTS_SERVICE_PID, service = TTSService.class)
+@ConfigurableService(category = "voice", label = "OpenAI TTS Service", description_uri = "voice:" + TTS_SERVICE_ID)
+
+@NonNullByDefault
+public class OpenAITTSService extends AbstractCachedTTSService {
+
+    private static final int REQUEST_TIMEOUT_MS = 10_000;
+    private final Logger logger = LoggerFactory.getLogger(OpenAITTSService.class);
+    private OpenAITTSConfiguration config = new OpenAITTSConfiguration();
+    private final HttpClient httpClient;
+    private final Gson gson = new Gson();
+    private static final Set<Voice> VOICES = Stream.of("nova", "alloy", "echo", "fable", "onyx", "shimmer")
+            .map(OpenAITTSVoice::new).collect(Collectors.toSet());
+
+    @Activate
+    public OpenAITTSService(@Reference HttpClientFactory httpClientFactory, @Reference TTSCache ttsCache,
+            Map<String, Object> config) {
+        super(ttsCache);
+        this.httpClient = httpClientFactory.getCommonHttpClient();
+    }
+
+    @Activate
+    protected void activate(Map<String, Object> config) {
+        this.config = new Configuration(config).as(OpenAITTSConfiguration.class);
+    }
+
+    @Modified
+    protected void modified(Map<String, Object> config) {
+        this.config = new Configuration(config).as(OpenAITTSConfiguration.class);
+    }
+
+    @Override
+    public Set<AudioFormat> getSupportedFormats() {
+        return Set.of(new AudioFormat(AudioFormat.CONTAINER_NONE, AudioFormat.CODEC_MP3, null, 16, 64000, 44100L));
+    }
+
+    @Override
+    public String getId() {
+        return TTS_SERVICE_ID;
+    }
+
+    @Override
+    public String getLabel(@Nullable Locale locale) {
+        return "OpenAI TTS Service";
+    }
+
+    @Override
+    public Set<Voice> getAvailableVoices() {
+        return VOICES;
+    }
+
+    /**
+     * Synthesizes the given text to audio data using the OpenAI API
+     * 
+     * @param text The text to synthesize
+     * @param voice The voice to use
+     * @param requestedFormat The requested audio format
+     * @return The synthesized audio data
+     * @throws TTSException If the synthesis fails
+     */
+    @Override
+    public AudioStream synthesizeForCache(String text, Voice voice, AudioFormat requestedFormat) throws TTSException {
+        JsonObject content = new JsonObject();
+        content.addProperty("model", config.model);
+        content.addProperty("input", text);
+        content.addProperty("voice", config.voice);
```

Comment:
It should synthesize text based on `voice.getUID()` otherwise it is ignoring the voice selected by the user.